modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -634,11 +634,33 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
       */
     CV_EXPORTS_W Net readNetFromCaffe(const String &prototxt, const String &caffeModel = String());
 
+    /** @brief Reads a network model stored in Caffe model in memory.
+      * @details This is an overloaded member function, provided for convenience.
+      * It differs from the above function only in what argument(s) it accepts.
+      * @param bufferProto buffer containing the content of the .prototxt file
+      * @param lenProto length of bufferProto
+      * @param bufferModel buffer containing the content of the .caffemodel file
+      * @param lenModel length of bufferModel
+      */
+    CV_EXPORTS_W Net readNetFromCaffe(const char *bufferProto, size_t lenProto,
```

Comment:
I had a problem with using it in python because of null characters in serialized protocol buffers. I think we should use just `CV_EXPORTS`.

Reply:
If we use only CV_EXPORTS, the function will not be exposed to Python/Java interface.

I should look at how the binding generator works so the function accept Python str instead of string

Reply:
It can't be exposed, because bindings don't work properly with native C++ pointers (at least they don't support buffers handling).

Reply:
In `modules/python/src2/hdr_parser.py`, I find a `CV_CARRAY` that seems to be able to wrap arrays. But the corresponding `#define` has disappeared from OpenCV headers.